plasma_framework/contracts/mocks/attackers/PaymentIFEInputChallengeAttacker.sol

```diff
@@ -0,0 +1,25 @@
+pragma solidity 0.5.11;
+pragma experimental ABIEncoderV2;
+
+import "../../src/exits/payment/routers/PaymentInFlightExitRouter.sol";
+import "../../src/exits/payment/routers/PaymentInFlightExitRouterArgs.sol";
+
+contract PaymentIFEInputChallengeAttacker {
+
+    bool private funded = false;
+    PaymentInFlightExitRouter private router;
+    PaymentInFlightExitRouterArgs.ChallengeInputSpentArgs private challengeArgs;
+
+    constructor(PaymentInFlightExitRouter _router, PaymentInFlightExitRouterArgs.ChallengeInputSpentArgs memory _challengeArgs) public {
+        router = _router;
+        challengeArgs = _challengeArgs;
+    }
+
+    /* solhint-disable no-complex-fallback */
+    function () external payable {
+        if (funded) {
+            router.challengeInFlightExitInputSpent(challengeArgs);
```

Comment:
All Payment*Attacker contracts could probably be replaced by a single contract that takes a generic address and encoded function selector with args. Here we would use `call`. I couldn't find a handy way to encode those structs on js side, so I proceeded with just copying the contracts. That's something that could be fixed in a follow-up PR.